File: desktopclient/src/main/java/com/goexp/galgame/gui/view/task/GameImgListTask.java
package com.goexp.galgame.gui.view.task;

import com.goexp.galgame.gui.db.mongo.Query;
import com.goexp.galgame.gui.model.Game;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.concurrent.Task;

import java.util.Optional;

import static com.mongodb.client.model.Filters.eq;

public class GameImgListTask extends Task<ObservableList<Game.GameImg>> {

    private int gameId;

    public GameImgListTask(int gameId) {
        this.gameId = gameId;

    }

    @Override
    protected ObservableList<Game.GameImg> call() {

        new Query.GameQuery.GameImgQuery();

        var g = Query.GameQuery.GameImgQuery.tlp.query()
                .where(eq(gameId))
                .one();
        return Optional.ofNullable(g.gameImgs)
                .map(FXCollections::observableArrayList)
                .orElse(FXCollections.emptyObservableList());
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/game/search/sub/GameImgListTask.scala
package com.goexp.galgame.gui.task.game.search.sub

import com.goexp.galgame.common.model.game.GameImg
import com.goexp.galgame.gui.db.mongo.query.GameImgQuery
import com.mongodb.client.model.Filters
import javafx.collections.{FXCollections, ObservableList}
import javafx.concurrent.Task

class GameImgListTask(private val gameId: Int) extends Task[ObservableList[GameImg]] {
  override protected def call: ObservableList[GameImg] = {
    GameImgQuery().where(Filters.eq(gameId)).one()
      .map {
        _.gameImgs
      }
      .map {
        FXCollections.observableArrayList(_)
      }
      .getOrElse(FXCollections.emptyObservableList[GameImg])

  }
}